python/pyarrow/tests/test_types.py

```diff
@@ -885,18 +885,32 @@ def test_type_id():
         assert isinstance(ty.id, int)
 
 
-def test_bit_width():
-    for ty, expected in [(pa.bool_(), 1),
-                         (pa.int8(), 8),
-                         (pa.uint32(), 32),
-                         (pa.float16(), 16),
-                         (pa.decimal128(19, 4), 128),
-                         (pa.decimal256(76, 38), 256),
-                         (pa.binary(42), 42 * 8)]:
-        assert ty.bit_width == expected
-    for ty in [pa.binary(), pa.string(), pa.list_(pa.int16())]:
+def test_bit_and_byte_width():
+    for ty, expected_bit_width, expected_byte_width in [
+        (pa.bool_(), 1, 0),
```

Comment:
Why would the boolean type return a byte width of 0? It should probably raise a ValueError.

Reply:
Yeah I was debating between returning 0 or an error, but you're right, I think raising an error may be better here.

Reply:
@kevinmingtarja Would you like to update this PR to address this comment?

Reply:
Hi @pitrou, I have just amended my commit to address this. Now, this will raise a ValueError.